pkg/kubectl/cmd/convert/register.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by client-gen. DO NOT EDIT.
+
+package convert
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	schema "k8s.io/apimachinery/pkg/runtime/schema"
+	serializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	admissionregistration "k8s.io/kubernetes/pkg/apis/admissionregistration/install"
+	apps "k8s.io/kubernetes/pkg/apis/apps/install"
+	auditregistration "k8s.io/kubernetes/pkg/apis/auditregistration/install"
+	authentication "k8s.io/kubernetes/pkg/apis/authentication/install"
+	authorization "k8s.io/kubernetes/pkg/apis/authorization/install"
+	autoscaling "k8s.io/kubernetes/pkg/apis/autoscaling/install"
+	batch "k8s.io/kubernetes/pkg/apis/batch/install"
+	certificates "k8s.io/kubernetes/pkg/apis/certificates/install"
+	coordination "k8s.io/kubernetes/pkg/apis/coordination/install"
+	core "k8s.io/kubernetes/pkg/apis/core/install"
+	events "k8s.io/kubernetes/pkg/apis/events/install"
+	extensions "k8s.io/kubernetes/pkg/apis/extensions/install"
+	networking "k8s.io/kubernetes/pkg/apis/networking/install"
+	policy "k8s.io/kubernetes/pkg/apis/policy/install"
+	rbac "k8s.io/kubernetes/pkg/apis/rbac/install"
+	scheduling "k8s.io/kubernetes/pkg/apis/scheduling/install"
+	settings "k8s.io/kubernetes/pkg/apis/settings/install"
+	storage "k8s.io/kubernetes/pkg/apis/storage/install"
+)
+
+var Scheme = runtime.NewScheme()
+var Codecs = serializer.NewCodecFactory(Scheme)
+var ParameterCodec = runtime.NewParameterCodec(Scheme)
+
+func init() {
+	v1.AddToGroupVersion(Scheme, schema.GroupVersion{Version: "v1"})
+	Install(Scheme)
+}
+
+// Install registers the API group and adds types to a scheme
+func Install(scheme *runtime.Scheme) {
```

Comment:
since the last dependency for internal client is gone in kubelet, the legacyscheme has to be initiated manually so that convert command can work correctly. (so i guess convert is also the last dependency for legacy scheme in kubectl too)

Reply:
question: can convert run w/o legacy scheme? i suppose it's impossible (for now)

Reply:
> (so i guess convert is also the last dependency for legacy scheme in kubectl too)

no it isnt. we are printing internal types in get command via legacy scheme.